### PR TITLE
[FIX] base: prevent break account hash

### PR DESCRIPTION
**Pull request by fmdl**
Description of the issue/feature this PR addresses:
Post a journal entry in a hashed journal and with a partner A. Merge A in partner B.
Print the hash report : it is now wrong.

Note: in master remove the partner in field to hash.

@oco-odoo 
@vin-odoo 

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
cc @epictete

File: odoo/addons/base/wizard/base_partner_merge.py [hunk 1]
```diff
@@ -324,9 +324,16 @@ def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
...
-        if extra_checks and 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([('partner_id', 'in', [partner.id for partner in src_partners])]):
+        if extra_checks and 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([('partner_id', 'in', [partner.id for partner in src_partners])], limit=1):
             raise UserError(_("Only the destination contact may be linked to existing Journal Items. Please ask the Administrator if you need to merge several contacts linked to existing Journal Items."))
 
+        # Prevent break the hash
+        if 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([
+            ('partner_id', 'in', [partner.id for partner in src_partners]),
+            ('parent_state', '=', 'posted'),
+            ('move_id.secure_sequence_number', '!=', 0)], limit=1):
+            raise UserError(_('You cannot merge a contact linked to a posted and hashed Journal Entry with another contact.'))
```
Comment (vin-odoo) [inline odoo/addons/base/wizard/base_partner_merge.py:335]:
```suggestion
        if 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([
            ('partner_id', 'in', src_partners.ids),
            ('parent_state', '=', 'posted'),
            ('move_id.secure_sequence_number', '!=', 0)], limit=1):
            raise UserError(_('You cannot merge a contact linked to a posted and hashed Journal Entry with another contact.'))
```
Don't forget to add the new translated term in the .pot file of the module.

I don't think checking if 'account.move.line' is in the env is useful at all, but as I see it done above I'll check with someone else to understand better why it is like that and come back to you later.

**Review by vin-odoo — CHANGES_REQUESTED** [review]

**Review by vin-odoo — CHANGES_REQUESTED** [review]
Could you add a test, please? Otherwise, it does enforce the rule in place in account.move.
I just wonder if the partner on the move itself shouldn't be in the hash, but it's not related to this pr

**Comment by fmdl** [comment]
@vin-odoo I will make the test.
yes the partner should not in the hash please see : https://github.com/odoo/odoo/pull/112804

**Pull request closed without merging**